iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/LoadTsfileAnalyzer.java

```diff
@@ -190,42 +187,50 @@ public Analysis analyzeFileByFile() {
   private void analyzeSingleTsFile(File tsFile) throws IOException, AuthException {
     try (final TsFileSequenceReader reader = new TsFileSequenceReader(tsFile.getAbsolutePath())) {
       // can be reused when constructing tsfile resource
-      final Map<String, List<TimeseriesMetadata>> device2TimeseriesMetadata =
-          reader.getAllTimeseriesMetadata(true);
-      if (device2TimeseriesMetadata.isEmpty()) {
-        LOGGER.warn("device2TimeseriesMetadata is empty, because maybe the tsfile is empty");
-        return;
-      }
-
-      // auto create or verify schema
-      if (IoTDBDescriptor.getInstance().getConfig().isAutoCreateSchemaEnabled()
-          || loadTsFileStatement.isVerifySchema()) {
-
-        final TimeSeriesIterator timeSeriesIterator =
-            new TimeSeriesIterator(tsFile, device2TimeseriesMetadata);
-        while (timeSeriesIterator.hasNext()) {
-          schemaAutoCreatorAndVerifier.autoCreateAndVerify(reader, timeSeriesIterator);
-        }
+      final TsFileSequenceReaderTimeseriesMetadataIterator timeseriesMetadataIterator =
+          new TsFileSequenceReaderTimeseriesMetadataIterator(reader, true);
 
-        schemaAutoCreatorAndVerifier.flushAndClearDeviceIsAlignedCacheIfNecessary();
-      }
+      long writePointCount = 0;
 
       // construct tsfile resource
       final TsFileResource tsFileResource = new TsFileResource(tsFile);
       if (!tsFileResource.resourceFileExists()) {
         // it will be serialized in LoadSingleTsFileNode
-        FileLoaderUtils.updateTsFileResource(device2TimeseriesMetadata, tsFileResource);
         tsFileResource.updatePlanIndexes(reader.getMinPlanIndex());
         tsFileResource.updatePlanIndexes(reader.getMaxPlanIndex());
       } else {
         tsFileResource.deserialize();
       }
 
+      // auto create or verify schema
+      if (IoTDBDescriptor.getInstance().getConfig().isAutoCreateSchemaEnabled()
+          || loadTsFileStatement.isVerifySchema()) {
+        while (timeseriesMetadataIterator.hasNext()) {
+          Map<String, List<TimeseriesMetadata>> device2TimeseriesMetadata =
+              timeseriesMetadataIterator.next();
+
+          // check if the tsfile is empty
+          if (device2TimeseriesMetadata.isEmpty()) {
+            LOGGER.warn("device2TimeseriesMetadata is empty, because maybe the tsfile is empty");
```

Comment:
move this logger before while.
if (!timeseriesMetadataIterator.hasNext())
LOGGER.warn("device2TimeseriesMetadata is empty, because maybe the tsfile is empty");